src/coordinator/block/column.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package block
+
+import (
+	"fmt"
+	"time"
+)
+
+// ColumnBlockBuilder builds a block optimized for column iteration
+type ColumnBlockBuilder struct {
+	block *columnBlock
+}
+
+type columnBlock struct {
+	columns [] column
+	meta    Metadata
+}
+
+// Meta returns the metadata for the block
+func (c *columnBlock) Meta() Metadata {
+	return c.meta
+}
+
+// StepIter returns a StepIterator
+func (c *columnBlock) StepIter() StepIter {
+	return &colBlockIter{
+		columns: c.columns,
+		meta:    c.meta,
+		index:   -1,
+	}
+}
+
+// TODO: allow series iteration
+// SeriesIter returns a SeriesIterator
+func (c *columnBlock) SeriesIter() SeriesIter {
+	return nil
+}
+
+// TODO: allow series iteration
+// SeriesMeta returns the metadata for each series in the block
+func (c *columnBlock) SeriesMeta() []SeriesMeta {
+	return nil
+}
+
+// Close frees up any resources
+// TODO: actually free up the resources
+func (c *columnBlock) Close() error {
+	return nil
+}
+
+type colBlockIter struct {
+	columns []column
+	meta    Metadata
+	index   int
+}
+
+// Next returns true if iterator has more values remaining
+func (c *colBlockIter) Next() bool {
+	c.index++
+	return c.index < len(c.columns)
+}
+
+// Current returns the current step
+func (c *colBlockIter) Current() Step {
+	col := c.columns[c.index]
+	t, err := c.meta.Bounds.TimeForIndex(c.index)
+	if err != nil {
+		panic(err)
```

Comment:
nit: add some additional text to panic for additional context

Reply:
added todo, will do that in the next diff